WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsParentViewModel.kt

```diff
@@ -1,10 +1,31 @@
 package org.wordpress.android.ui.bloggingprompts
 
+import androidx.annotation.VisibleForTesting
 import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.model.SiteModel
+import javax.inject.Inject
 
-class BloggingPromptsParentViewModel : ViewModel() {
-    fun onOpen() = Unit
+@HiltViewModel
+class BloggingPromptsParentViewModel @Inject constructor(
+    private val handle: BloggingPromptsSiteProvider,
+    private val analyticsTracker: BloggingPromptsAnalyticsTracker,
+) : ViewModel() {
+    fun start(site: SiteModel) {
+        handle.setSite(site)
+    }
 
-    @Suppress("UnusedPrivateMember")
-    fun onSectionSelected(promptSection: PromptSection) = Unit
+    fun onOpen(currentTab: PromptSection) {
+        analyticsTracker.trackScreenAccessed(getSite(), currentTab)
+    }
+
+    fun onSectionSelected(currentTab: PromptSection) {
+        analyticsTracker.trackTabSelected(getSite(), currentTab)
+    }
+
+    @VisibleForTesting
+    internal fun getSite(): SiteModel = checkNotNull(handle.getSite()) {
```

Comment:
> I couldn't find this method being called directly in the unit tests. If so, would it make sense to make it private and remove the @VisibleForTesting annotation? 🤔

Another miss when I refactored the tests later. Will make the changes. 

> handle it differently, like showing the error state for example?

Agreed. I'm a little embarrassed I let the crash happen.

That makes me wonder though. **Should we show an error screen (blocking the user if there is analytics related issues) or let it silently fail with logging (ensuring the feature can still be accessed)?** We can choose to either compromise on the accuracy of analytics events or the user experience. This is of course assuming the rest of the pages do not require the site argument which it might (leading to the same error screen). Will keep this in mind for the next PR around networking layer.

Reply:
- Removed the @VisibleForTesting  
- Removed the runtime exception. If siteModel is missing, tracking will silently fail. 

Note: I couldn't add logging because AppLog needs an enum as a TAG and I couldn't add BloggingPrompts since it's in a different library. Any suggestions?

Reply:
> Should we show an error screen (blocking the user if there is analytics related issues) or let it silently fail with logging (ensuring the feature can still be accessed)?

> This is of course assuming the rest of the pages do not require the site argument which it might (leading to the same error screen)

Good point. I believe this will become clearer when the network layer is implemented, but IMHO we should not let the screen show an error just because of the analytics tracking. I was assuming that this `SiteModel` would be used by the main logic too, but I might be wrong 🙂 

> I couldn't add logging because AppLog needs an enum as a TAG and I couldn't add BloggingPrompts since it's in a different library. Any suggestions?

Hmm did you try something like [this](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/util/WPActivityUtils.java#L51)?